src/models/contract.py

```diff
@@ -0,0 +1,104 @@
+import typing
+
+import sqlalchemy as sa
+from sqlalchemy.future import select
+from sqlalchemy.orm.session import Session
+
+from src.models import Base
+
+
+class Contract(Base):
+    """Contract model."""
+
+    __tablename__ = "contracts"
+
+    id = sa.Column(sa.String, primary_key=True)
```

Comment:
`Contract` model requires an `address` field. With this setup now we cannot have 2 contracts with the same address on 2 different chains (e.g. ETH, BSC). A contract can therefore be uniquely identifiable by a (`address`, `chain`) tuple.

Reply:
I've added the address field.  I have not had time to change the interfaces to use a (`address`, `chain`) as a primary key for the contracts.  They are currently using the `id` field which is an auto-incrementing integer.  I can change this tomorrow.